goth/runner/probe.py

```diff
@@ -237,30 +246,59 @@ def set_agent_env_vars(self, env: Dict[str, str]) -> None:
             }
         )
 
-    def run_command_on_host(
+    @contextlib.asynccontextmanager
+    async def run_command_on_host(
         self,
         command: str,
         env: Optional[Dict[str, str]] = None,
-        timeout: int = 300,
-    ) -> asyncio.Task:
+        command_timeout: float = 300,
+    ) -> Iterator[Tuple[asyncio.Task, PatternMatchingWaitableMonitor]]:
         """Run `command` on host in given `env` and with optional `timeout`.
 
-        The command is run the environment extending `env` with variables needed
+        The command is run in the environment extending `env` with variables needed
         to communicate with the daemon running in this probe's container.
 
-        Returns the `asyncio` task that logs output from the command. The task
-        can be awaited in order to wait until the command completes.
-        """
+        Internally, this method used `process.run_command()` to run `command`.
```

Comment:
```suggestion
        Internally, this method uses `process.run_command()` to run `command`.
```

Reply:
So many typos, thanks for catching them!